File: mario_less.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>

int main(void)
{
    int height;
    int space;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 0 || height > 23);

    space=height - 1;
    int star = 2;

    for (int i = 2; i < height + 2; i++)
    {

        for (int j = 0; j < space; j++)
        {
            printf(" ");
        }

        for (int k = 0; k < star; k++)
        {
            printf("#");
        }
        space -= 1;
        star += 1;
        printf("\n");
    }

    return 0;
}

File: mario_less.py
from cs50 import get_int


def mario():
    # ask user about height in case when input is less than 0 or greater than 23 repeat until input will be proper
    while True:
        height = get_int("Height: ")
        if (height > -1) and (height < 24):
            break
    # variable rowLength store how many charachter are in each row
    row_length = height + 1
    how_many_hash = 2
    for row in range(height):
        how_many_space = row_length - how_many_hash
        # printing spaces
        for number in range(how_many_space):
            print(" ", end="")
        # printing hashes
        for number in range(how_many_hash):
            print("#", end="")
        # increment number of hashes
        how_many_hash += 1
        print()


mario()
